### PR TITLE
Support Affinity for ContainerOps

### LINKED ISSUE #1884: Missing Affinity for Containerop
**What happened:**
currently we are missing function to setup the affinity for containerop. 
Am gonna create a PR to support that shortly

### PR DESCRIPTION
**Pull request by hamedhsn**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1886)

Fix #1884

### CODE REVIEW

**Comment by Ark-kun** [comment]
/ok-to-test

In future we'll probably move all Kubernetes pod attributes (e.g. labels, annotations, tolerations etc) to `ContainerOp.pod_options.`

File: sdk/python/tests/compiler/compiler_tests.py [hunk 1]
```diff
@@ -548,6 +549,24 @@ def test_tolerations(self):
...
+
+    op1 = dsl.ContainerOp(
+      name='download',
+      image='busybox',
+      command=['sh', '-c'],
+      arguments=['sleep 10; wget localhost:5678 -O /tmp/results.txt'],
+      file_outputs={'downloaded': '/tmp/results.txt'}) \
+      .add_affinity(affinity)
+
+    self._test_op_to_template_yaml(op1, file_base_name='affinity')
```
Comment (Ark-kun) [inline sdk/python/tests/compiler/compiler_tests.py]: Can you please do a feature behavior test here similar to `test_set_retry` (as opposed to just comparing the whole YAML)? https://github.com/kubeflow/pipelines/blob/1fd29cd235ee0ce50b3ca8907f094cc5c5546e99/sdk/python/tests/compiler/compiler_tests.py#L347..L358

Comment (hamedhsn) [reply sdk/python/tests/compiler/compiler_tests.py]: Added.

**Comment by hamedhsn** [comment]
@Ark-kun Added another test, looks good now.

File: sdk/python/tests/compiler/compiler_tests.py [hunk 2]
```diff
@@ -548,6 +579,24 @@ def test_tolerations(self):
 
     self._test_op_to_template_yaml(op1, file_base_name='tolerations')
 
+  def test_affinity(self):
```
Comment (Ark-kun) [inline sdk/python/tests/compiler/compiler_tests.py]: You can remove this duplicate test now.

File: sdk/python/tests/compiler/testdata/affinity.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# Copyright 2019 Google LLC
```
Comment (Ark-kun) [inline sdk/python/tests/compiler/testdata/affinity.yaml]: You can remove this file now.

File: sdk/python/tests/compiler/compiler_tests.py (hunk 2 above)
Comment (hamedhsn) [reply sdk/python/tests/compiler/compiler_tests.py]: done

File: sdk/python/tests/compiler/testdata/affinity.yaml (hunk 3 above)
Comment (hamedhsn) [reply sdk/python/tests/compiler/testdata/affinity.yaml]: done

**Comment by hamedhsn** [comment]
@Ark-kun have a look pls. should be fine now.

**Pull request merged**